prepare-videochat.sh

```diff
@@ -395,34 +478,63 @@ set +e
 
 #sudo v4l2loopback-ctl set-caps $GST_0_10_VIDEO_CAPS $DEVICE
 
-"$GSTLAUNCH" -e -vt --gst-plugin-spew \
-             --gst-debug="$GST_DEBUG" \
-  souphttpsrc location="$VIDEO_URL" do-timestamp=true is-live=true \
+pipeline_video() {
+  echo souphttpsrc location="$VIDEO_URL" do-timestamp=true is-live=true \
     ! multipartdemux \
     ! jpegdec \
     $GST_FLIP \
     ! $GST_VIDEO_CONVERTER \
     ! videoscale \
     ! videorate \
     ! $GST_VIDEO_CAPS \
-    ! v4l2sink device="$DEVICE" sync=true \
-  souphttpsrc location="$AUDIO_URL" do-timestamp=true is-live=true \
+    ! v4l2sink device="$DEVICE" sync=true
+}
+
+pipeline_audio() {
+  echo souphttpsrc location="$AUDIO_URL" do-timestamp=true is-live=true \
     ! $GST_AUDIO_CAPS ! queue \
-    ! pulsesink device="$SINK_NAME" sync=true \
+    ! pulsesink device="$SINK_NAME" sync=true
+}
+
+if [ $CAPTURE_STREAM = av ]; then
+    PIPELINE="$( pipeline_audio )  $( pipeline_video )"
+elif [ $CAPTURE_STREAM = a ]; then
+    PIPELINE=$( pipeline_audio )
+elif [ $CAPTURE_STREAM = v ]; then
+    PIPELINE=$( pipeline_video )
+else
+    error "Incorrect CAPTURE_STREAM value ($CAPTURE_STREAM). Should be a, v or av."
+fi
+
+# echo "$PIPELINE"
+
+"$GSTLAUNCH" -e -vt --gst-plugin-spew \
+             --gst-debug="$GST_DEBUG" \
+   $PIPELINE \
     >feed.log 2>&1 &
+    # Maybe we need edit this pipeline to transfer it to "Monitor of IP Webcam" to be able to use it as a microphone?
 
 GSTLAUNCH_PID=$!
 
-info "IP Webcam video is streaming through v4l2loopback device $DEVICE.
-IP Webcam audio is streaming through pulseaudio sink '$SINK_NAME'.
-You can now open your videochat app."
+if [ $CAPTURE_STREAM = av ]; then
+   MESSAGE="IP Webcam audio is streaming through pulseaudio sink '$SINK_NAME'.\nIP Webcam video is streaming through v4l2loopback device $DEVICE.\n"
+elif [ $CAPTURE_STREAM = a ]; then
+    MESSAGE="IP Webcam audio is streaming through pulseaudio sink '$SINK_NAME'.\n"
+elif [ $CAPTURE_STREAM = v ]; then
+    MESSAGE="IP Webcam video is streaming through v4l2loopback device $DEVICE.\n"
+else
+    error "Incorrect CAPTURE_STREAM value ($CAPTURE_STREAM). Should be a, v or av."
+fi
+info "$MESSAGE You can now open your videochat app."
 
 echo "Press enter to end stream"
 perl -e '<STDIN>'
 
 kill $GSTLAUNCH_PID > /dev/null 2>&1 || echo ""
+stop_adb
```

Comment:
Not sure if this is a good idea - the user might be using adb for something else on the side.